pkg/client/didexchange/client.go

```diff
@@ -133,11 +136,17 @@ func (c *Client) QueryConnections(request *QueryConnectionsParams) ([]*QueryConn
 	}, nil
 }
 
-// QueryConnectionByID fetches single connection record for given id
-func (c *Client) QueryConnectionByID(id string) (*QueryConnectionResult, error) {
-	// TODO sample response, to be implemented as part of #226
+// GetConnection fetches single connection record for given id
+func (c *Client) GetConnection(connectionID string) (*QueryConnectionResult, error) {
+	conn, err := c.recorder.GetConnection(connectionID)
+	if err != nil {
+		if errors.Is(err, storage.ErrDataNotFound) {
+			return nil, ErrConnectionNotFound
+		}
+		return nil, fmt.Errorf("cannot fetch state from store: connectionid=%s err=%s", connectionID, err)
+	}
 	return &QueryConnectionResult{
```

Comment:
`QueryConnectionResult` is a verbose name for a connection record :).

Is this struct embedding persistence.ConnectionData?

Reply:
Rename to ConnectionResult